flexwrfoutput/postprocess.py

```diff
@@ -0,0 +1,102 @@
+"""
+Additional functions needed in preprocess to secure compatibility to xWRF
+"""
+from datetime import datetime
+
+import numpy as np
+import xarray as xr
+
+
+def _prepare_conc_units(ds: xr.Dataset) -> xr.Dataset:
+    """
+    Change units of footprint data to be pint compatible.
+    """
+    ds.CONC.attrs["units"] = (
+        "s m^3 kg^-1" if ds.CONC.units == "s m3 kg-1" else ds.CONC.units
+    )
+    return ds
+
+
+def _make_attrs_consistent(ds: xr.Dataset) -> xr.Dataset:
+    """
+    Change attribute names and values of FLEXPART-WRF output to be compatible with xWRF
+        postprocess.
+    """
+
+    ndims_lat = ds.dims["south_north"]
+    ndims_lon = ds.dims["west_east"]
+
+    lat_center_index = (
+        int(ndims_lat // 2)
+        if ndims_lat % 2 == 1
+        else [int(ndims_lat // 2), int(ndims_lat // 2 - 1)]
+    )
+    lon_center_index = (
+        int(ndims_lon // 2)
+        if ndims_lon % 2 == 1
+        else [int(ndims_lon // 2), int(ndims_lon // 2 - 1)]
+    )
+
+    ds.attrs["CEN_LAT"] = (
+        ds.XLAT.isel(south_north=lat_center_index, west_east=lon_center_index)
+        .mean()
+        .values
+    )
+    ds.attrs["CEN_LON"] = (
+        ds.XLONG.isel(south_north=lat_center_index, west_east=lon_center_index)
+        .mean()
+        .values
+    )
+    ds.attrs["MOAD_CEN_LAT"] = ds.attrs["CEN_LAT"]
+
+    return ds
+
+
+def _prepare_coordinates(ds: xr.Dataset) -> xr.Dataset:
+    """
+    Set useful coordinates.
+    """
+    # if created with flexwrfinput z dim corresponds to z_stag of WRF
+    ds = ds.rename_dims({"bottom_top": "bottom_top_stag"})
+    ds = ds.assign_coords(z_height=("bottom_top_stag", ds.ZTOP.values))
+    ds.z_height.attrs = dict(description="Top of layer (above surface)", units="m")
+    # Set measurement times as coordinate for releases
+    unformatted_simulation_start = str(ds.SIMULATION_START_DATE) + str(
+        ds.SIMULATION_START_TIME
+    ).zfill(6)
```

Comment:
Bist du dir sicher, dass die Simulation time nie auch Minuten beinhalten wird? In WRF kann man zumindest auch Minuten als start angeben.

Reply:
`SIMULATION_START_TIME` ist eine Integer, mit den Stellen `hhmmss`. Wenn also führende Größen 0 sind dann fehlen die Stellen einfach und ich fülle die fehlenden Nullen wieder auf.